packages/wallet/src/communication/index.ts

```diff
@@ -0,0 +1,56 @@
+import { Commitment } from '../domain';
+import { messageRelayRequested } from 'magmo-wallet-client';
+
+export enum Strategy {
+  IndirectFunding = 'IndirectFunding',
+}
+
+export interface BaseProcessAction {
+  processId: string;
+  type: string;
+}
+
+// FUNDING
+export const STRATEGY_PROPOSED = 'WALLET.FUNDING.STRATEGY_PROPOSED';
+export interface StrategyProposed extends BaseProcessAction {
+  type: typeof STRATEGY_PROPOSED;
+  strategy: Strategy;
+}
+export const strategyProposed = (processId: string, strategy): StrategyProposed => ({
+  type: STRATEGY_PROPOSED,
+  processId,
+  strategy,
+});
+
+export const STRATEGY_APPROVED = 'WALLET.FUNDING.STRATEGY_APPROVED';
+export interface StrategyApproved extends BaseProcessAction {
+  type: typeof STRATEGY_APPROVED;
+}
+export const strategyApproved = (processId: string): StrategyApproved => ({
+  type: STRATEGY_APPROVED,
+  processId,
+});
+
+// CONCLUDING
+export const CONCLUDE_CHANNEL = 'WALLET.CONCLUDING.CONCLUDE_CHANNEL';
+export interface ConcludeChannel extends BaseProcessAction {
+  type: typeof CONCLUDE_CHANNEL;
+  commitment: Commitment;
+  signature: string;
+}
+export const concludeChannel = (
+  processId: string,
+  commitment: Commitment,
+  signature: string,
+): ConcludeChannel => ({
+  type: CONCLUDE_CHANNEL,
+  processId,
+  commitment,
+  signature,
+});
+
+export type Message = StrategyProposed | StrategyApproved | ConcludeChannel;
+export function sendMessage(to: string, message: Message) {
```

Comment:
Do you think it's true that this will basically always be called with one of the methods above? If so, it might be worth skipping the middleman and defining `sendConcludeChannel` etc.

Reply:
It should only be called with those methods above, yes.